File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let answer = 0;

function solution(numbers, target) {
    const length = numbers.length;
    
    recursive(0, 0, numbers, length, target);
    
    return answer;
    
}


// 재귀함수
function recursive(count, sum, arr, length, target) {
    
    if(count === length) {
        if (target === sum) {
            answer++;
        }
        return;
    }
    
    recursive(count + 1, sum + arr[count], arr, length, target);
    recursive(count + 1, sum - arr[count], arr, length, target);
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    bfs = [0]
    
    for number in numbers:
        temp = []
        
        for element in bfs:
            temp.append(element + number)
            temp.append(element - number)
        
        bfs = temp
    
    for element in bfs:
        if element == target:
            answer += 1
            
    
    return answer